datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -1128,10 +1128,18 @@ fn finalize_aggregation(
         | AggregateMode::Single
         | AggregateMode::SinglePartitioned => {
             // merge the state to the final value
-            accumulators
+            let res = accumulators
                 .iter()
-                .map(|accumulator| accumulator.evaluate().map(|v| v.to_array()))
-                .collect::<Result<Vec<ArrayRef>>>()
+                .map(|accumulator| {
+                    let column_value = accumulator.evaluate_v2();
```

Comment:
@alamb 
If we want `accumulator.evaluate()` here, we need to introduce something like `ScalarValue::Array(Option<Vec<ScalarValue>>)`, the Vec<> here is to represent values of rows.

Reply:
It seems not a good idea to introduce `ScalarValue:Array`
https://github.com/apache/arrow-datafusion/blob/2ec0bc106540e4cbebd8b7b7269cbf3fa8ae34e3/datafusion/common/src/scalar.rs#L60-L61